core_lib_generator/config_collectors/data_access.py

```diff
@@ -11,18 +11,29 @@ def generate_data_access_template(db_entities: list) -> list:
     def is_exists_data_access(user_input: str):
         return is_exists(user_input, data_access)
 
+    entity_list = []
     if db_entities:
         for entity in db_entities:
-            entity_name = entity['key']
-            db_conn = entity['db_connection']
+            name = entity['key']
+            conn = entity['db_connection']
+            entity_list.append(f'{name} -> {conn}')
+        add_da = True
+        while add_da:
+            entity_input = (input_list(entity_list, 'Select the Entity to create DataAccess for')).split(' -> ')

```

Comment:
The idea of adding an array with a separator and splitting it every time.
It's Not Resilient to changes. 
please try to work with index.

suggestion:
```
entity_list = []
connection_index = entity_list.index(entity_input)
if connection_index > -1: # case selected. 
    selected_entity = db_entities[connection_index]
    if entity.get('key') == selected_entity. key and ntity.get('db_connection') == selected_entity.db_connection: